mongodbatlas/resource_mongodbatlas_cluster_migrate.go

```diff
@@ -0,0 +1,343 @@
+										Default:  0,
+									},
+								},
+							},
+						},
+						"zone_name": {
+							Type:     schema.TypeString,
+							Optional: true,
+							Default:  "ZoneName managed by Terraform",
+						},
+					},
+				},
+			},
+			"mongo_db_version": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"mongo_uri": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"mongo_uri_updated": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"mongo_uri_with_options": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"paused": {
+				Type:     schema.TypeBool,
+				Computed: true,
+			},
+			"srv_address": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"state_name": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"advanced_configuration": {
+				Type:     schema.TypeMap,
+				Optional: true,
+				Computed: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"fail_index_key_too_long": {
+							Type:     schema.TypeBool,
+							Optional: true,
+							Computed: true,
+						},
+						"javascript_enabled": {
+							Type:     schema.TypeBool,
+							Optional: true,
+							Computed: true,
+						},
+						"minimum_enabled_tls_protocol": {
+							Type:     schema.TypeString,
+							Optional: true,
+							Computed: true,
+						},
+						"no_table_scan": {
+							Type:     schema.TypeBool,
+							Optional: true,
+							Computed: true,
+						},
+						"oplog_size_mb": {
+							Type:     schema.TypeInt,
+							Optional: true,
+							Computed: true,
+						},
+						"sample_size_bi_connector": {
+							Type:     schema.TypeInt,
+							Optional: true,
+							Computed: true,
+						},
+						"sample_refresh_interval_bi_connector": {
+							Type:     schema.TypeInt,
+							Optional: true,
+							Computed: true,
+						},
+					},
+				},
+			},
+			"labels": {
+				Type:     schema.TypeSet,
+				Optional: true,
+				Set: func(v interface{}) int {
+					var buf bytes.Buffer
+					m := v.(map[string]interface{})
+					buf.WriteString(m["key"].(string))
+					buf.WriteString(m["value"].(string))
+					return hashcode.String(buf.String())
+				},
+				Computed: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"key": {
+							Type:     schema.TypeString,
+							Optional: true,
+							Computed: true,
+						},
+						"value": {
+							Type:     schema.TypeString,
+							Optional: true,
+							Computed: true,
+						},
+					},
+				},
+			},
+			"snapshot_backup_policy": computedCloudProviderSnapshotBackupPolicySchema(),
+		},
+	}
+}
+
+func resourceMongoDBAtlasClusterStateUpgradeV0(rawState map[string]interface{}, meta interface{}) (map[string]interface{}, error) {
+
+	log.Println("[INFO] Found MongoDB Cluser state v0; migrating to v1")
+	log.Printf("%v", rawState["advanced_configuration"])
+
+	//rawState["advanced_configuration"] = []interface{}{}
```

Comment:
could you remove the logs and the comment plz?

Reply:
sure, I just leave the first lo message is important

Reply:
Info Is OK